### PR TITLE
Add glob pattern support for .agentsweepignore

### LINKED ISSUE #125: enhancement: glob pattern support in .agentsweepignore for path-based suppression
You will add a fourth line form, `path:<glob>`, to `.agentsweepignore` so a whole path or directory can suppress findings.

## How it fits together

```mermaid
flowchart TD
    A["ignore.py<br/>IgnoreSet.add_line() (line 35)"] --> B["store path:<glob><br/>in self.globs"]
    C["ignore.py<br/>IgnoreSet.matches() (line 50)"] --> D["fnmatch the finding relpath<br/>against each glob"]
    E["pipeline._scan_file() (line 787)<br/>already computes relpath (line 805)"] --> C
    D --> F["finding suppressed"]
```

`matches()` currently takes `(rule, value, fp)`. Give it the finding's relpath too. The only caller is `_scan_file()` in `pipeline.py` (call site around line 815), and `relpath` is already in scope there.

## Steps

1. Add `self.globs: list[str] = []` to `IgnoreSet.__init__` (ignore.py, line 26) and make `__bool__` (line 32) count it.
2. In `add_line()` (line 35), parse a leading `path:` and store the rest as a glob.
3. In `matches()` (line 50), accept a `relpath` argument and return True when `fnmatch.fnmatch(relpath, glob)` matches any stored glob. `fnmatch` or `PurePath.match` semantics both work.
4. Update the caller in `pipeline._scan_file()` (call site around line 815) to pass `relpath`.
5. Document the new form in the module docstring at the top of `ignore.py`.
6. Add tests in `tests/test_ignore.py`: a glob that matches a finding's path suppresses it, one that does not match leaves it alone, and a glob combined with an existing `rule:` or fingerprint line in the same file works.

## Files

- `src/agentsweep/ignore.py` — `IgnoreSet` (line 25), module docstring (line 1)
- `src/agentsweep/pipeline.py` — `_scan_file()` (line 787)
- `tests/test_ignore.py`

## How to claim

Comment "I'll take this", fork the repo, branch from `main`, and open a PR that links this issue.

### PR DESCRIPTION
**Pull request by itsnotming**
## What & why

This PR adds glob pattern support for `.agentsweepignore` using `path:` entries.

It allows findings to be ignored based on relative file paths in addition to existing rule, fingerprint, and value matching.

Closes #125

## Type of change

- [ ] Bug fix
- [ ] New agent source
- [ ] New detection rule
- [x] Feature / enhancement
- [ ] Refactor / docs / chore

## Testing

```
uv run pytest tests/test_ignore.py -v
62 passed

uv run pytest
608 passed
```

## Checklist

- [ ] `pytest` passes locally, and CI is green on **all** platforms (Linux **and** Windows, Python 3.11–3.13)
- [ ] `mypy src/` is clean — it runs on every matrix leg and is the most common reason a PR goes red
- [x] No new runtime dependency, or the PR explains why one is needed (the core install is deliberately three packages)
- [x] No real secrets, tokens, or raw history-file contents are committed — tests use synthetic, non-live examples
- [x] Redaction / write-path changes preserve the corruption-prevention invariants (path containment, atomic replace, mandatory no-clobber backup, post-write validation, line-count preservation, audit trail)
- [ ] **New detection rule?** Added to `RULES` **and** `ROTATION_GUIDANCE` **and** a fixture/test, with every regex quantifier bounded (no unbounded `.*`), and the keyword pre-filter kept lossless
- [ ] **New source?** Registered in `SOURCES` with process markers, a menu entry, and a round-trip test (see CONTRIBUTING.md → "Adding a new source — checklist")
- [x] `--json` and non-tty output stay machine-clean (no banners / styling, no `ui` import on those paths)
- [x] Did not re-introduce `force-include` in `pyproject.toml`

## Summary by CodeRabbit

* **New Features**
  * Added support for `path:<pattern>` entries in `.agentsweepignore`.
  * Path-based rules can suppress findings in matching relative file paths.
  * Suppression results now report the number of findings ignored by matching paths.
  * Supports path matching across common path separator formats.

* **Documentation**
  * Added guidance and examples for path glob patterns, including matching behavior.

* **Tests**
  * Added coverage for matching and non-matching path-based ignore rules.

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds `path:<pattern>` as a fourth entry format in `.agentsweepignore`, enabling users to suppress findings across entire sets of files using `fnmatch` glob patterns. The implementation correctly handles Windows path separator normalization and deduplicates patterns via a `set`.

- **`ignore.py`**: Adds a `globs: set[str]` field, parses `path:` entries in `add_line`, and applies them via `fnmatch.fnmatch` in `matches`, with `replace("\\", "/")` for Windows compat.
- **`pipeline.py`**: Threads the pre-computed `relpath` into `ignores.matches()` at the single call site.
- **`tests/test_ignore.py`**: Adds unit tests for classification, boolean truthiness, Windows separators, negative-match assertions, and two end-to-end integration tests in `TestPathGlobSuppression`.
</details>

<h3>Confidence Score: 4/5</h3>

- Safe to merge with a small follow-up fix for the fingerprint/glob ordering edge case.
- All previously-identified gaps (Windows normalization, set vs list, bool coverage, negative-match tests, end-to-end integration test) have been resolved. One remaining issue: the fingerprint heuristic fires before the `path:` prefix check, so a glob like `path:2023:logs` is silently added to `fingerprints` instead of `globs`, making the new feature non-functional for that pattern class. The fix is a reordering of two `if` blocks in `add_line`.
- `src/agentsweep/ignore.py` — the ordering of the fingerprint heuristic vs the `path:` prefix check in `add_line`.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/agentsweep/ignore.py | Core feature implementation: adds `globs` set, parses `path:` entries, applies fnmatch matching with Windows normalization. The fingerprint check runs before the `path:` check, so `path:<digit>:<anything>` is misclassified as a fingerprint, but this edge case is explicitly tested and accepted by the author. |
| src/agentsweep/pipeline.py | Minimal one-line change to thread `relpath` into `ignores.matches()`. Correct and safe — `relpath` is already computed above the loop. |
| tests/test_ignore.py | Good coverage added: classification, bool truthiness, Windows separator test, negative-match assertion, and two end-to-end integration tests in TestPathGlobSuppression. |
| README.md | Documents the new `path:` entry type with accurate fnmatch semantics, a practical example, and a clarifying note about `**` not being recursive. |

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["add_line(raw)"] --> B{starts with rule:?}
    B -- Yes --> C[rules.add]
    B -- No --> D{3-part fingerprint?\nrsplit colon 2, middle is digit}
    D -- Yes --> E[fingerprints.add]
    D -- No --> F{starts with path:?}
    F -- Yes --> G{glob empty?}
    G -- No --> H[globs.add]
    G -- Yes --> I[silently skip]
    F -- No --> J[values.add]

    K["matches(rule, value, fp, relpath)"] --> L{rule in rules?}
    L -- Yes --> M[return True]
    L -- No --> N{fp in fingerprints?}
    N -- Yes --> M
    N -- No --> O{value in values?}
    O -- Yes --> M
    O -- No --> P{"any fnmatch(relpath.replace('\\\\','/'), glob)\nfor glob in globs"}
    P -- Yes --> M
    P -- No --> Q[return False]
```
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=%23%23%23%20Issue%201%0Asrc%2Fagentsweep%2Fignore.py%3A52-61%0A**%60path%3A%60%20check%20silently%20loses%20when%20glob%20starts%20with%20%60%3Cdigit%3E%3A%3Ctoken%3E%60**%0A%0AThe%20fingerprint%20heuristic%20%28%60rsplit%28%22%3A%22%2C%202%29%60%20%2B%20%60parts%5B1%5D.isdigit%28%29%60%29%20runs%20before%20the%20%60path%3A%60%20prefix%20check%2C%20so%20a%20line%20like%20%60path%3A2023%3Alogs%60%20is%20parsed%20as%20three%20parts%20%60%5B%22path%22%2C%20%222023%22%2C%20%22logs%22%5D%60%2C%20%60%222023%22.isdigit%28%29%60%20is%20True%2C%20and%20the%20entry%20lands%20in%20%60fingerprints%60%20instead%20of%20%60globs%60.%20The%20user's%20glob%20pattern%20is%20silently%20discarded.%20This%20affects%20any%20pattern%20where%20the%20first%20segment%20of%20the%20glob%20%28before%20the%20first%20%60%3A%60%29%20is%20a%20pure%20integer%20%E2%80%94%20e.g.%2C%20%60path%3A1%3A*%60%20or%20%60path%3A2024%3Aarchive%60.%20Moving%20the%20%60path%3A%60%20check%20above%20the%20fingerprint%20block%20removes%20the%20ambiguity%2C%20at%20the%20cost%20that%20a%20pre-existing%20fingerprint%20for%20a%20file%20literally%20named%20%60path%60%20%28i.e.%2C%20%60path%3A1%3Asome-rule%60%29%20would%20now%20be%20treated%20as%20a%20glob%3B%20that%20trade-off%20is%20almost%20certainly%20correct%20since%20%60path%60%20as%20a%20bare%20filename%20is%20far%20less%20common%20than%20explicit%20%60path%3A%60%20glob%20entries.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=ishannaik%2Fagent-sweep&pr=163&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=6">Fix All in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22ishannaik%2Fagent-sweep%22%20on%20the%20existing%20branch%20%22add-glob-pattern-support%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22add-glob-pattern-support%22.%0A%0A%23%23%23%20Issue%201%0Asrc%2Fagentsweep%2Fignore.py%3A52-61%0A**%60path%3A%60%20check%20silently%20loses%20when%20glob%20starts%20with%20%60%3Cdigit%3E%3A%3Ctoken%3E%60**%0A%0AThe%20fingerprint%20heuristic%20%28%60rsplit%28%22%3A%22%2C%202%29%60%20%2B%20%60parts%5B1%5D.isdigit%28%29%60%29%20runs%20before%20the%20%60path%3A%60%20prefix%20check%2C%20so%20a%20line%20like%20%60path%3A2023%3Alogs%60%20is%20parsed%20as%20three%20parts%20%60%5B%22path%22%2C%20%222023%22%2C%20%22logs%22%5D%60%2C%20%60%222023%22.isdigit%28%29%60%20is%20True%2C%20and%20the%20entry%20lands%20in%20%60fingerprints%60%20instead%20of%20%60globs%60.%20The%20user's%20glob%20pattern%20is%20silently%20discarded.%20This%20affects%20any%20pattern%20where%20the%20first%20segment%20of%20the%20glob%20%28before%20the%20first%20%60%3A%60%29%20is%20a%20pure%20integer%20%E2%80%94%20e.g.%2C%20%60path%3A1%3A*%60%20or%20%60path%3A2024%3Aarchive%60.%20Moving%20the%20%60path%3A%60%20check%20above%20the%20fingerprint%20block%20removes%20the%20ambiguity%2C%20at%20the%20cost%20that%20a%20pre-existing%20fingerprint%20for%20a%20file%20literally%20named%20%60path%60%20%28i.e.%2C%20%60path%3A1%3Asome-rule%60%29%20would%20now%20be%20treated%20as%20a%20glob%3B%20that%20trade-off%20is%20almost%20certainly%20correct%20since%20%60path%60%20as%20a%20bare%20filename%20is%20far%20less%20common%20than%20explicit%20%60path%3A%60%20glob%20entries.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=ishannaik%2Fagent-sweep&pr=163&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=6">Fix All in Codex</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=%23%23%23%20Issue%201%0Asrc%2Fagentsweep%2Fignore.py%3A52-61%0A**%60path%3A%60%20check%20silently%20loses%20when%20glob%20starts%20with%20%60%3Cdigit%3E%3A%3Ctoken%3E%60**%0A%0AThe%20fingerprint%20heuristic%20%28%60rsplit%28%22%3A%22%2C%202%29%60%20%2B%20%60parts%5B1%5D.isdigit%28%29%60%29%20runs%20before%20the%20%60path%3A%60%20prefix%20check%2C%20so%20a%20line%20like%20%60path%3A2023%3Alogs%60%20is%20parsed%20as%20three%20parts%20%60%5B%22path%22%2C%20%222023%22%2C%20%22logs%22%5D%60%2C%20%60%222023%22.isdigit%28%29%60%20is%20True%2C%20and%20the%20entry%20lands%20in%20%60fingerprints%60%20instead%20of%20%60globs%60.%20The%20user's%20glob%20pattern%20is%20silently%20discarded.%20This%20affects%20any%20pattern%20where%20the%20first%20segment%20of%20the%20glob%20%28before%20the%20first%20%60%3A%60%29%20is%20a%20pure%20integer%20%E2%80%94%20e.g.%2C%20%60path%3A1%3A*%60%20or%20%60path%3A2024%3Aarchive%60.%20Moving%20the%20%60path%3A%60%20check%20above%20the%20fingerprint%20block%20removes%20the%20ambiguity%2C%20at%20the%20cost%20that%20a%20pre-existing%20fingerprint%20for%20a%20file%20literally%20named%20%60path%60%20%28i.e.%2C%20%60path%3A1%3Asome-rule%60%29%20would%20now%20be%20treated%20as%20a%20glob%3B%20that%20trade-off%20is%20almost%20certainly%20correct%20since%20%60path%60%20as%20a%20bare%20filename%20is%20far%20less%20common%20than%20explicit%20%60path%3A%60%20glob%20entries.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&pr=163&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursor.svg?v=6">Fix All in Cursor</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
### Issue 1
src/agentsweep/ignore.py:52-61
**`path:` check silently loses when glob starts with `<digit>:<token>`**

The fingerprint heuristic (`rsplit(":", 2)` + `parts[1].isdigit()`) runs before the `path:` prefix check, so a line like `path:2023:logs` is parsed as three parts `["path", "2023", "logs"]`, `"2023".isdigit()` is True, and the entry lands in `fingerprints` instead of `globs`. The user's glob pattern is silently discarded. This affects any pattern where the first segment of the glob (before the first `:`) is a pure integer — e.g., `path:1:*` or `path:2024:archive`. Moving the `path:` check above the fingerprint block removes the ambiguity, at the cost that a pre-existing fingerprint for a file literally named `path` (i.e., `path:1:some-rule`) would now be treated as a glob; that trade-off is almost certainly correct since `path` as a bare filename is far less common than explicit `path:` glob entries.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
`````

</details>

<sub>Reviews (4): Last reviewed commit: ["Address review feedback for path ignore ..."](https://github.com/ishannaik/agent-sweep/commit/30c975b650b90006aa4a33bb966647b616426eff) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=47617691)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

**Comment by Ishannaik** [comment]
Thanks for this, and sorry to send it back rather than merge. The structure is right: the `path:` prefix parses cleanly, `__bool__` picks up globs, `matches()` takes relpath with a default so nothing else breaks, and you wrote tests. Two things need fixing first.

**1. The issue link points at the wrong issue.**

The description says `Closes #12`, but #12 was a `list-sources` issue that merged a while back. This implements **#125**. Worth swapping so the right issue closes on merge.

**2. `fnmatch` does not mean what the example pattern implies.**

Your test uses `path:**/fixtures/**` and passes, which hides the problem. `fnmatch` has no concept of `**`, and its single `*` already crosses `/`. So:

```python
>>> fnmatch.fnmatch('tests/fixtures/x.jsonl', '**/fixtures/**')
True     # passes, but only because * spans separators anyway
>>> fnmatch.fnmatch('fixtures/x.jsonl', '**/fixtures/**')
False    # a top-level fixtures/ dir silently does not match
>>> fnmatch.fnmatch('tests/fixtures/x.jsonl', '*.jsonl')
True     # matches nested paths, which gitignore would not
```

Someone writes `path:**/fixtures/**`, sees it work on a nested directory, then hits a repo where fixtures sits at the root and gets no suppression and no error. Silent non-suppression is a bad failure mode for a tool people run to find secrets.

Either fix works:

- Keep `fnmatch` and document it. Add a line to the docstring at the top of `ignore.py` and to the README section saying these are `fnmatch` patterns where `*` crosses `/`, and change the example to `path:*/fixtures/*`. Add a test asserting the top-level case behaves however you decide.
- Or switch to `PurePath(relpath).match(glob)`, which handles `**` the way people expect from gitignore. Costs a `PurePath` construction per finding, which is nothing next to the regex pass.

I lean toward the second, since `.agentsweepignore` sits next to `.gitignore` in a user's head and matching that intuition beats matching stdlib.

Ping me when it's pushed and I'll re-review. Keeping this open, the work is most of the way there.

Discord if you want to talk it through: https://discord.gg/KKvtRhQvRv

**Review by Ishannaik — COMMENTED** [review]
The wiring through `pipeline.py` is correct. Five suggestions before merge:

1. `globs` should be `set[str]`, not `list[str]`. Matches `rules`/`fingerprints`/`values` and prevents silent duplicates when root + cwd ignore files merge.
2. Module docstring lists valid line types. Add `path:`.
3. `fnmatch` treats `**` the same as `*` (no recursive gitignore semantics) and is case-insensitive on Windows. Add a docstring note.
4. A file literally named `path` (e.g. `path:1:rule-id`) gets misparsed as a glob since `path:` is checked before the fingerprint branch.
5. Add a negative-match assertion, a `__bool__`-with-globs test, and a CLI integration test. The other suppression modes (`rule:`, fingerprint, literal value) each have one.

(1) and (5) prevent real bugs. The rest are polish.

We started a Discord for agent-sweep, CloakBin, and warp: https://discord.gg/KKvtRhQvRv
A star helps other people find the project.

**Comment by itsnotming** [comment]
Thanks for the detailed review!

I've updated the PR based on your suggestions:

1. documented the `fnmatch` semantics for `path:` entries in both `ignore.py` and the README
2. replaced the misleading `**/fixtures/**` example with `*/fixtures/*`
3. added positive and negative unit tests for path glob matching
4. added end-to-end tests covering path-based suppression
5. all tests pass locally (`611 passed`)

**Comment by itsnotming** [comment]
I addressed the requested changes:
1. changed globs to a set
2. normalized Windows path separators before matching
3. added tests for __bool__, Windows paths, and the path:1:rule fingerprint edge case
4. updated the parser ordering so fingerprints take precedence over the path: prefix
5. The full test suite passes locally (614 passed).

**Comment by Ishannaik** [comment]
This addresses what I raised last time. Documenting the `fnmatch` semantics rather than switching to `PurePath` is a fine call, and the README now says plainly that `*` crosses `/`, so nobody hits the silent non-match I was worried about. The Windows separator normalization on `relpath` is a good addition I had not asked for.

Three nits, none blocking.

**Normalization is one-sided.** You normalize `relpath` but not the pattern, so a rule written with Windows separators never matches:

```python
ig.add_line('path:*\\fixtures\\*')
ig.matches('aws-access-key', 'AKIA...', 'fp', 'tests/fixtures/sample.jsonl')  # False
```

A Windows user writing the pattern the way their paths look gets silence. Same `.replace("\\", "/")` on the glob closes it.

**The fingerprint heuristic shadows some path patterns.** It runs before the `path:` branch, so `path:*:1:*` lands in `fingerprints` and never becomes a glob. Verified. Rare shape, but worth either parsing `path:` first or noting the limitation next to the docs.

**README example overlaps.** `path:*/fixtures/*` already covers the file used in the fingerprint example, so that example does not demonstrate a distinct form.

Fix the first one and I will merge. The other two are your call. Tests look genuine, they assert routing, matching, non-matching and the e2e payload rather than just not crashing.

https://discord.gg/KKvtRhQvRv if you want to discuss.

**Review by Ishannaik — APPROVED** [review]
## Independent review — APPROVE (nits)

Verified locally on branch ddc47ec: full suite passes (614 on this base), test_ignore.py 68 passed, mypy clean, no new ruff findings.

All five review asks are in: globs is a set, docstring updated, fingerprint branch parses before path: (with a regression test for a file named "path"), and the negative/bool/CLI tests exist. Greptile's Windows-separator P1 is fixed by normalizing backslashes in matches() with a dedicated test.

Two small CodeRabbit items remain, both non-blocking: the README fingerprint example sits under path:*/fixtures/* which already swallows it, and there is no glob+rule+fingerprint combination test.

Housekeeping: this branch is 28 commits behind main and the fork's CI runs are waiting on first-contributor approval — rebase and approve so the current 664-test suite runs green before merge.

**Comment by Ishannaik** [comment]
Your glob support is approved and verified: 614 tests pass on the branch, all five review asks are in, and the Windows-separator fix works. Rebase onto main and it is ready to merge.

First contribution to agent-sweep, so while you are here: a star on the repo helps more than you would think, and new good-first-issues land in the community chat before they get labeled.

https://discord.gg/KKvtRhQvRv

**Review by Ishannaik — CHANGES_REQUESTED** [review]
Re-tested on `30c975b`. The Windows separator problem is gone and both `set` comments are addressed, so this is close. One thing left.

major `src/agentsweep/ignore.py:50` - the fingerprint heuristic runs before the `path:` check, so any glob whose first two segments look like `<digit>:<token>` is silently filed as a fingerprint and never matches a path:

```python
>>> s = IgnoreSet(); s.add_line("path:2024:logs/*")
>>> s.globs, s.fingerprints
(set(), {'path:2024:logs/*'})
```

The user gets no error. The line sits in their `.agentsweepignore` looking correct and suppresses nothing, which is the worst failure mode for an ignore file. Fix: move the `if line.startswith("path:")` block above the `rsplit(":", 2)` heuristic. An explicit prefix should win over a guess. I checked the reorder against fingerprints, `rule:` lines and bare literals and none of them change:

```
'path:2024:logs/*'                glob=['2024:logs/*']
'tests/a.jsonl:12:aws-access-key' fp=['tests/a.jsonl:12:aws-access-key']
'rule:jwt'                        rule=['jwt']
'sk-literal'                      val=['sk-literal']
```

Verified as already fixed, so ignore the bot threads still showing them open:

- Windows separators work. `path:*/fixtures/*` matches both `tests/fixtures/a.jsonl` and `tests\fixtures\a.jsonl` now.
- `globs` is a `set` and uses `.add`.

The README section is accurate. Spelling out that `*` crosses `/` and that `**` means nothing here is the right call, because `fnmatch` surprises everyone who assumes gitignore semantics.

VERDICT: NEEDS-CHANGES
TESTS: 672 pass and `mypy src/` is clean on this head. `tests/test_ignore.py` covers parsing and positive matches; nothing covers a `path:` line that fails to parse, which is exactly the case above. Worth one test alongside the fix.

Push the reorder and I will merge.

https://discord.gg/KKvtRhQvRv

File: src/agentsweep/ignore.py [hunk 1]
```diff
@@ -27,15 +28,21 @@ def __init__(self) -> None:
         self.rules: set[str] = set()
         self.fingerprints: set[str] = set()
         self.values: set[str] = set()
+        self.globs: list[str] = []
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply src/agentsweep/ignore.py]: Fixed on `30c975b`: `self.globs: set[str] = set()`.

File: src/agentsweep/ignore.py [hunk 2]
```diff
@@ -27,15 +28,21 @@ def __init__(self) -> None:
...
+        return bool(self.rules or self.fingerprints or self.values or self.globs)
 
     def add_line(self, raw: str) -> None:
         line = raw.strip()
         if not line or line.startswith("#"):
             return
+        if line.startswith("path:"):
+            glob = line[len("path:"):].strip()
+            if glob:
+                self.globs.append(glob)
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply src/agentsweep/ignore.py]: Fixed, it uses `self.globs.add(glob)` now.

File: src/agentsweep/ignore.py [hunk 3]
```diff
@@ -47,10 +58,12 @@ def add_line(self, raw: str) -> None:
         else:
             self.values.add(line)
 
-    def matches(self, rule: str, value: str, fp: str) -> bool:
+    def matches(self, rule: str, value: str, fp: str, relpath: str = "") -> bool:
         return (rule in self.rules
                 or fp in self.fingerprints
-                or value in self.values)
+                or value in self.values
+                or any(fnmatch.fnmatch(relpath, glob) for glob in self.globs)
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply src/agentsweep/ignore.py]: Fixed. `matches()` now does `relpath.replace("\\", "/")`, and I confirmed `path:*/fixtures/*` matches both `tests/fixtures/a.jsonl` and `tests\fixtures\a.jsonl`.

File: src/agentsweep/ignore.py [hunk 4]
```diff
@@ -27,30 +32,45 @@ def __init__(self) -> None:
...
             self.fingerprints.add(line)
-        else:
-            self.values.add(line)
+            return
+
+        if line.startswith("path:"):
+            glob = line[len("path:"):].strip()
+            if glob:
+                self.globs.add(glob)
+            return
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply src/agentsweep/ignore.py:61]: Confirmed and still open, this is the one blocking finding. `IgnoreSet().add_line("path:2024:logs/*")` leaves `globs` empty and files it under `fingerprints`. Detail and a checked fix are in the review.

File: tests/test_ignore.py [hunk 5]
```diff
@@ -179,6 +187,16 @@ def test_value_match_suppresses_by_secret_content(self):
...
 
+    def test_path_glob_matches_relpath(self):
+        ig = IgnoreSet()
+        ig.add_line("path:**/fixtures/**")
+        assert ig.matches(
+            "aws-access-key",
+            AWS_KEY,
+            "tests/fixtures/sample.jsonl:1:aws-access-key",
+            "tests/fixtures/sample.jsonl",
+        )
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply tests/test_ignore.py:212]: Agreed and worth doing alongside the parser fix. A `path:` line that silently fails to parse is the specific case with no coverage.

File: README.md [hunk 6]
```diff
@@ -437,10 +438,16 @@ Example:
 AKIAIOSFODNN7EXAMPLE
 # ignore the whole slack-webhook rule
 rule:slack-webhook
+# ignore findings in nested fixture directories
+path:*/fixtures/*
 # one exact false positive in a fixture
 tests/fixtures/claude/sample.jsonl:42:aws-access-key-id
 ```
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply README.md:451]: Leaving as is. The README carries one `path:` example and explicitly documents that `*` crosses `/` and `**` means nothing here, which is the surprise worth spelling out.

File: tests/test_ignore.py [hunk 7]
```diff
@@ -104,6 +104,14 @@ def test_rule_prefix_strips_whitespace(self):
...
         ig.add_line("rule: aws-access-key ")
         assert "aws-access-key" in ig.rules
+        
+    def test_path_prefix_goes_to_globs(self):
+        ig = IgnoreSet()
+        ig.add_line("path:**/fixtures/**")
+        assert "**/fixtures/**" in ig.globs
+        assert not ig.rules
+        assert not ig.fingerprints
+        assert not ig.values
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply tests/test_ignore.py:114]: Agreed, and it lines up with the blocking parser finding. The coverage worth adding is a `path:` line that fails to parse, plus a negative match, in the same push as the reorder fix.

**Comment by Ishannaik** [comment]
Merged. Verified by running rather than reading:

```
$ python -m pytest -q
672 passed          # 670 on origin/main

$ agentsweep scan --root $TMP/root --stats     # no ignore file
  [3/5] ✘ FINDINGS  2 secret(s) in 2 file(s)

$ echo 'path:*/fixtures/*' > .agentsweepignore && agentsweep scan --root $TMP/root --stats
  ⚠ 1 finding(s) suppressed by .agentsweepignore
  [3/5] ✘ FINDINGS  1 secret(s) in 1 file(s)
```

Also checked the two things most likely to be wrong and neither was: `*` really does cross `/` as the README now says, and `path:1:aws-access-key` really does classify as a fingerprint because the rsplit check runs first. You codified that second one in a test, which is the right call. Ruff count is 16 on your branch and 16 on main, so no new lint debt.

One thing worth a follow-up, not a blocker. `matches()` takes `relpath: str = ""`, and an empty relpath makes `path:*` match everything:

```python
>>> ig = IgnoreSet(); ig.add_line("path:*")
>>> ig.matches("some-rule", "AKIA...", "f:1:r")   # relpath omitted
True
```

`pipeline.py:952` is the only production caller and it passes `relpath`, so nothing is broken today. But this is a secret scanner, where the failure mode is silence, and a default that means "suppress everything" is the wrong direction to fail. The default only exists so the 13 existing test call sites keep compiling.

Want to take that as a second PR? Drop the default, update the test call sites to pass a path explicitly. Small, and it puts the safe behaviour in the type signature.
